### PR TITLE
deprecating types after colon

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: compiler/frontend/src/org/jetbrains/kotlin/types/expressions/BasicExpressionTypingVisitor.java [hunk 1]
```diff
@@ -227,7 +228,7 @@ private void checkForCastImpossibilityOrRedundancy(
                 expression.getLeft(), context.dataFlowInfo, actualType, context);
         for (JetType possibleType : possibleTypes) {
             if (typeChecker.isSubtypeOf(possibleType, targetType)) {
-                context.trace.report(USELESS_CAST_STATIC_ASSERT_IS_FINE.on(expression));
+                context.trace.report(USELESS_CAST.on(expression));
```
Comment (udalov) [inline compiler/frontend/src/org/jetbrains/kotlin/types/expressions/BasicExpressionTypingVisitor.java]: Probably this shouldn't be reported at all here?

Consider this example:
```
trait A
trait B
trait C : A, B

fun foo(a: A) {}
fun foo(b: B) {}

fun bar(c: C) = foo(c as A)   // <-- incorrect warning here, the cast is not useless
```

Comment (abreslav) [inline compiler/frontend/src/org/jetbrains/kotlin/types/expressions/BasicExpressionTypingVisitor.java]: Agree

Comment (yole) [inline compiler/frontend/src/org/jetbrains/kotlin/types/expressions/BasicExpressionTypingVisitor.java]: @udalov Will do, thanks

**Comment by yole** [comment]
Merged manually.

**Pull request closed without merging**